### ISSUE TITLE
alter table `files` add `folder_path` varchar(255) null - ER_BAD_FIELD_ERROR: Unknown column '`project_name`.`t0`.`formats`' in 'CHECK'

### ISSUE THREAD

**Issue by abanchev** [strapi/strapi#13986]
### Required System information

- Node.js version: v14.19.3
- NPM version: 6.14.17
- Strapi version: 4.3.2
- Database: mysql
- Operating system: ubuntu

### Describe the bug

After updating dependencies from 4.2.3 to 4.3.2 strapi fails to start.
```
error: alter table `files` add `folder_path` varchar(255) null - ER_BAD_FIELD_ERROR: Unknown column '`project_name`.`t0`.`formats`' in 'CHECK'
Error: ER_BAD_FIELD_ERROR: Unknown column '`project_name`.`t0`.`formats`' in 'CHECK'
    at Query.Sequence._packetToError (/home/gitlab-runner/project_name_api/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
    at Query.ErrorPacket (/home/gitlab-runner/project_name_api/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
    at Protocol._parsePacket (/home/gitlab-runner/project_name_api/node_modules/mysql/lib/protocol/Protocol.js:291:23)
    at Parser._parsePacket (/home/gitlab-runner/project_name_api/node_modules/mysql/lib/protocol/Parser.js:433:10)
    at Parser.write (/home/gitlab-runner/project_name_api/node_modules/mysql/lib/protocol/Parser.js:43:10)
    at Protocol.write (/home/gitlab-runner/project_name_api/node_modules/mysql/lib/protocol/Protocol.js:38:16)
    at Socket.<anonymous> (/home/gitlab-runner/project_name_api/node_modules/mysql/lib/Connection.js:88:28)
    at Socket.<anonymous> (/home/gitlab-runner/project_name_api/node_modules/mysql/lib/Connection.js:526:10)
    at Socket.emit (events.js:400:28)
    at Socket.emit (domain.js:475:12)
    --------------------
    at Protocol._enqueue (/home/gitlab-runner/project_name_api/node_modules/mysql/lib/protocol/Protocol.js:144:48)
    at Connection.query (/home/gitlab-runner/project_name_api/node_modules/mysql/lib/Connection.js:198:25)
    at /home/gitlab-runner/project_name_api/node_modules/knex/lib/dialects/mysql/index.js:132:18
    at new Promise (<anonymous>)
    at Client_MySQL._query (/home/gitlab-runner/project_name_api/node_modules/knex/lib/dialects/mysql/index.js:126:12)
    at executeQuery (/home/gitlab-runner/project_name_api/node_modules/knex/lib/execution/internal/query-executioner.js:37:17)
    at Client_MySQL.query (/home/gitlab-runner/project_name_api/node_modules/knex/lib/client.js:146:12)
    at /home/gitlab-runner/project_name_api/node_modules/knex/lib/execution/transaction.js:363:24
    at new Promise (<anonymous>)
    at Client_MySQL.trxClient.query (/home/gitlab-runner/project_name_api/node_modules/knex/lib/execution/transaction.js:358:12)
    
```

### Steps to reproduce the behavior

1. Install 4.2.3
2. Update deps to 4.3.2
3. Start project

### Expected behavior

Migration should complete.

**Comment by abanchev** [comment]
Originally it was using strapi 4.1.11 -> 4.2.0 -> 4.2.2 -> 4.2.3 -> 4.3.2
just last update had any issues

**Comment by derrickmehaffy** [comment]
Can you please make sure you delete the lock file and reinstall your node modules. It looks like you have some cached packages.

**Comment by abanchev** [comment]
During the CI process  node_modules, build and cache folders are deleted already, both locks are deleted (yarn for dev, npm for prod) as well.
I just reverted back to 4.2.3 and it is working good there.
The issue seems to have something to do with the new folder paths support in the upload plugin.
This is the package lock:
```
"@strapi/plugin-i18n": "4.3.2",
    "@strapi/plugin-users-permissions": "4.3.2",
    "@strapi/provider-email-amazon-ses": "4.3.2",
    "@strapi/provider-upload-aws-s3": "ifchy/strapi-provider-upload-aws-s3-with-cdn",
    "@strapi/strapi": "4.3.2",
    "better-sqlite3": "7.5.3",
    "mysql": "^2.18.1"
```

The only difference from a standard install is a custom s3 provider(that allows configurable file paths), but I just checked there haven't been any changes to that package since 4.2.3 so the issue should not be related.
